src/averageAges.js

```diff
@@ -55,10 +69,32 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const children = people.filter(
+    child => {
+      const mother = people.find(person => child['mother'] === person['name']);
+
+      return onlyWithSon
+        ? mother && child['sex'] === 'm'
+        : mother;
+    }
+  );
+
+  const ageDiff = children.reduce(
+    (sum, child) => sum + (
+      child['born'] - people.find(
+        mother => mother['name'] === child['mother']
```

Comment:
If you want to use dynamic keys - you should create constants for storing their names, but it's redundant here